File: CrossFiles/PHP2/Class2.php
<?php

class Class2 {
    private $input;
    private $db;

    public function __construct($input) {
        $this->input = $input;
        $this->db = null;
    }

    public function connectDb($dbName) {
        try {
            $this->db = new PDO("sqlite:$dbName");
            $this->db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
            return true;
        } catch(PDOException $e) {
            echo "Connection failed: " . $e->getMessage();
            return false;
        }
    }

    public function process() {
        echo "Class2 processing: {$this->input}\n";
        if (!$this->db) {
            echo "Database not connected\n";
            return;
        }

        // Potentially unsafe operation
        $query = "SELECT * FROM users WHERE username = '{$this->input}'";
        echo "Executing query: $query\n";

        try {
            $stmt = $this->db->query($query);
            $results = $stmt->fetchAll(PDO::FETCH_ASSOC);
            print_r($results);
        } catch(PDOException $e) {
            echo "Query failed: " . $e->getMessage();
        }
    }

    public function __destruct() {
        $this->db = null;
    }
}
?>

File: CrossFiles/GoModules/internal/class2/class2.go
package class2

import (
	"database/sql"
	"fmt"

	_ "github.com/mattn/go-sqlite3"
)

type Class2 struct {
	input string
	db    *sql.DB
}

func New(input string) *Class2 {
	return &Class2{input: input}
}

func (c *Class2) ConnectDb(dbName string) error {
	var err error
	c.db, err = sql.Open("sqlite3", dbName)
	if err != nil {
		return fmt.Errorf("failed to connect to database: %v", err)
	}
	return nil
}

func (c *Class2) Process() {
	fmt.Printf("Class2 processing: %s\n", c.input)
	if c.db == nil {
		fmt.Println("Database not connected")
		return
	}

	query := fmt.Sprintf("SELECT * FROM users WHERE username = '%s'", c.input)
	fmt.Printf("Executing query: %s\n", query)

	rows, err := c.db.Query(query)
	if err != nil {
		fmt.Printf("Error executing query: %v\n", err)
		return
	}
	defer rows.Close()

	for rows.Next() {
		var username string
		if err := rows.Scan(&username); err != nil {
			fmt.Printf("Error scanning row: %v\n", err)
			return
		}
		fmt.Printf("Username: %s\n", username)
	}
}

func (c *Class2) CloseDb() {
	if c.db != nil {
		c.db.Close()
	}
}
